drivers/serial/Kconfig.stm32

```
@@ -25,18 +25,24 @@ config UART_STM32
 
 if UART_STM32
 
-if SOC_STM32U575XX || SOC_STM32U585XX || \
-		   SOC_STM32H562XX || SOC_STM32H563XX || SOC_STM32H573XX
+if \
+	SOC_STM32U575XX || SOC_STM32U585XX || \
+	SOC_STM32H503XX || \
+	SOC_STM32H562XX || SOC_STM32H563XX || SOC_STM32H573XX || \
+	SOC_STM32H523XX || SOC_STM32H533XX
 
 choice UART_STM32U5_ERRATA_DMAT
 	prompt "Workaround for DMAT errata on selected devices"
 	default UART_STM32U5_ERRATA_DMAT_LOWPOWER if PM
 	default UART_STM32U5_ERRATA_DMAT_NOCLEAR if !PM
 	help
 	  Handles erratum "USART does not generate DMA requests after
-	  setting/clearing DMAT bit".
-	  Seen in Errata Sheet 0499 § 2.19.2 and §2.20.1 for stm32u57x/u58x,
-	          Errata Sheet 0565 § 2.14.1 and §2.15.1 for stm32h56x/h57x
+	  setting/clearing DMAT bit" as described in the errata sheets:
+
+	  - ES0499 §2.21.2 and §2.22.1 for STM32U575xx/U585xx
+	  - ES0561 §2.11.2 and §2.12.1 for STM32H503xx
+	  - ES0565 §2.16.2 and §2.17.1 for STM32H562xx/H563xx/H573xx
+	  - ES0621 §2.14.2 and §2.15.1 for STM32H523xx/H533xx
```

I would suggest to mention the ES revision number, for consistency of the section references.
By the way, character `§` is not a standard ASCII char. I would suggest to replace it with explicit `section` word.
```suggestion
	  - ES0499 rev. 10, sections 2.21.2 and 2.22.1 for STM32U575xx/U585xx
	  - ES0561 rev. 2, sections 2.11.2 and 2.12.1 for STM32H503xx
	  - ES0565 rev. 7, sections 2.16.2 and 2.17.1 for STM32H562xx/H563xx/H573xx
	  - ES0621 rev. 2, sections 2.14.2 and 2.15.1 for STM32H523xx/H533xx